### ISSUE TITLE
[BUG] --cython-directives flag to build_ext/old_build_ext is broken

### ISSUE THREAD

**Issue by segevfiner** [cython/cython#4441]
**Describe the bug**
Using `python setup.py build_ext --cython-directives=linetrace=True` with `build_ext`/`old_build_ext` is broken.

**To Reproduce**
Just run `python setup.py build_ext --cython-directives=linetrace=True` on any Cython project using the `build_ext`/`old_build_ext`

**Expected behavior**
It should work, instead it raises an exception because `linetrace=True` is left as a string and not parsed into a list of directives. (How to parse the value part might require some consideration...)

**Environment (please complete the following information):**
 - OS: Windows
 - Python version 3.10.0
 - Cython version 0.29.24

**Additional context**
See https://groups.google.com/g/cython-users/c/C52mc9lm5xo

**Comment by scoder** [comment]
Right, that looks wrong. I guess no-one has ever used this. That means that we can probably leave this unresolved, given that we're deprecating `old_build_ext` in favour of `new_build_ext`. Let's just make sure it works there.

**Comment by scoder** [comment]
For the latest changes in `new_build_ext`, see https://github.com/cython/cython/pull/4339

**Comment by segevfiner** [comment]
Hmmm. It doesn't seem that even in that PR we have a way to pass `cython-directives` from either the CLI or from the `Extension` instance or from any setup keyword? Nor any other Cython specific flags to cythonize?

**Comment by segevfiner** [comment]
Welp... Looks like even passing `cython_directives` to the `Extension` in `old_build_ext` is broken. Just passing it to the normal extension generates a warning, so it looks like you need to import `Extension` from `Cython.Distutils`, but then it just errors out with `error: unknown file type '.pyx' (from 'cypcap.pyx')`

**Comment by matusvalo** [comment]
BTW part of #3541 is migrating all directives to `new_build_ext`. This should close this issue. I can work on that when #4339 is merged to master.
